### PR TITLE
179

### LINKED ISSUE #179: Large Feed Warnings
Use a warning dialog to warn the user before rendering a large feed. The definition of "large" varies based on browser. Look at implementation from previous map/list for a starting point.

### PR DESCRIPTION
**Pull request by hasimpson-usgs**
fixes #179 

Add large feed warnings.  
Still need to add DownloadView to showClientMaxError.  Will open new ticket for that.

### CODE REVIEW

File: src/htdocs/js/latesteqs/LatestEarthquakes.js [hunk 1]
```diff
@@ -234,6 +235,14 @@ var LatestEarthquakes = function (options) {
     });
   };
 
+  _this.revertToDefaultFeed = function () {
+    var toSet;
+
+    _config.options.feed.selectById('1day_m25');
+    toSet = {'feed': _config.options.feed.getSelected()};
```
Comment (ehunter-usgs) [inline src/htdocs/js/latesteqs/LatestEarthquakes.js]: You can do this more simply:
```
_this.model.set({
	‘feed’: _config.options.feed.get('1day_m25')
});
```

File: src/htdocs/js/latesteqs/Catalog.js [hunk 2]
```diff
@@ -214,6 +353,68 @@ var Catalog = function (options) {
...
+      title: 'Error',
+      closable: false,
+      classes: ['modal-error', 'catalog'],
+      destroyOnHide: true
+    });
+
+    message.innerHTML = [
+      '<p>',
+        'The current selection includes ', data.count, ' earthquakes, ',
+        'which is more than is allowed.',
```
Comment (ehunter-usgs) [inline src/htdocs/js/latesteqs/Catalog.js]: Can we say what is "more than is allowed", use data.maxAllowed?

File: src/htdocs/js/latesteqs/Catalog.js [hunk 3]
```diff
@@ -126,6 +237,54 @@ var Catalog = function (options) {
...
+      _this.showServerMaxError(data);
+      return;
+    } else if (count > _maxResults) {
+      _this.showClientMaxError(_this.loadQuery, data);
+    } else {
+      _this.loadQuery();
+    }
+  };
+
+  _this.onFeedChange = function () {
```
Comment (ehunter-usgs) [inline src/htdocs/js/latesteqs/Catalog.js]: _this.onFeedChange() exists twice in this class, delete one.

File: src/htdocs/js/latesteqs/Catalog.js [hunk 4]
```diff
@@ -47,6 +59,40 @@ var Catalog = function (options) {
...
+    // if event does not exist in the new collection, update model
+    if (!_this.get(eq.id)) {
+      _this.model.set({
+        'event': null
+      });
+    }
+  };
+
+  _this.checkSearchLimit = function (url, data) {
+    _this.trigger('loading');
```
Comment (ehunter-usgs) [inline src/htdocs/js/latesteqs/Catalog.js]: In the old interface we have a loading icon that indicates a search is being loaded. Right now the interface feels broken while this search is being loaded.

File: src/htdocs/js/latesteqs/Catalog.js [hunk 5]
```diff
@@ -214,6 +353,68 @@ var Catalog = function (options) {
...
+
+    dialog = ModalView(message, {
+      title: 'Caution',
+      closable: false,
+      classes: ['modal-warning', 'catalog'],
+      buttons: [
+        {
+          callback: function () { dialog.hide(); callback(data);},
+          classes:['Footer', 'continue'],
+          text: 'Continue anyway'
```
Comment (ehunter-usgs) [inline src/htdocs/js/latesteqs/Catalog.js]: Don't forget `destroyOnHide: true` unless you are explicitly destroying the ModalView elsewhere. It looks like showServerMaxError() is including this option (see line 400).

File: src/htdocs/js/latesteqs/Catalog.js [hunk 6]
```diff
@@ -74,6 +120,55 @@ var Catalog = function (options) {
...
+          '<button class="button-as-link revert">',
+            'Show Realtime Data Instead',
+          '</button>',
+          '<small class="catalog-action-description">',
+            '1 Day, Magnitude 2.5+ Worldwide',
+          '</small>',
+        '</p>'
+      ].join('');
+
+    p.querySelector('.revert').addEventListener('click',
```
Comment (ehunter-usgs) [inline src/htdocs/js/latesteqs/Catalog.js]: Make sure to remove all event listeners. That means you should create a method that this click event calls.

File: src/htdocs/js/latesteqs/Catalog.js (hunk 4 above)
Comment (hasimpson-usgs) [inline src/htdocs/js/latesteqs/Catalog.js]: Added a simple message,  but haven't added the spinner.  There's a ticket for that.

**Comment by hasimpson-usgs** [comment]
Refactored dialog code into seperate class.  Currently need to double click to get events to work.   
Also need to refactor FeedWarningView further.  Use a base class for things like getDialogXXX, and then seperate classes and/or child classes for all the dialogs.

**Comment by ehunter-usgs** [comment]
This has been replaced by https://github.com/usgs/earthquake-latest-earthquakes/pull/232

**Pull request closed without merging**